kamu-adapter-http/src/smart_protocol/messages.rs

```diff
@@ -147,7 +147,24 @@ pub enum ObjectPushStrategy {
     HttpUpload,
 }
 
-/// Push stage 3: complete handshake indication
+/// Push stage 3: object upload progress
+#[derive(Serialize, Deserialize, Clone, PartialEq, Eq, Debug)]
+pub struct DatasetPushObjectsUploadInProgress {
```

Comment:
I don't really like using this as PING substitute.
- On a poor connection a single large enough object might upload slow enough for connection to be cut even before there's any change in progress to report;
- I highly doubt that any more "sophisticated" payload will have higher chances to prevent idle connection drop than sending a single byte of data like ping.

How I imagine ideal keep-alive is:
- Server may at any point send PING message to the client
- Client has to reply with PONG as soon as possible

This way:
- Server is in charge of ping frequency (it can be adjusted centrally to always be below idle timeout of an LB)
- It's a much simpler rule to follow for client-side protocol implementors - it doesn't complicate the state machine as much as upload progress does.

Reply:
I see now that you have `min_upload_progress_delay_sec` that would solve "single large object" problem, but I still like the idea of server being in charge of the ping frequency better.

Reply:
Addressed